client/consensus/babe/rpc/src/lib.rs

```diff
@@ -126,22 +131,20 @@ impl<B, C, SC> BabeApi for BabeRpcHandler<B, C, SC>
 
 			let mut claims: HashMap<AuthorityId, EpochAuthorship> = HashMap::new();
 
-			let key_pairs = {
-				let keystore = keystore.read();
-				epoch.authorities.iter()
-					.enumerate()
-					.flat_map(|(i, a)| {
-						keystore
-							.key_pair::<sp_consensus_babe::AuthorityPair>(&a.0)
-							.ok()
-							.map(|kp| (kp, i))
-					})
-					.collect::<Vec<_>>()
-			};
+			let keys = epoch.authorities.iter()
+				.enumerate()
+				.filter_map(|(i, a)| {
+					if keystore.read().has_keys(&[(a.0.to_raw_vec(), AuthorityId::ID)]) {
```

Comment:
I am not sure if `keystore.read()` is cheap, but if it is not, it should probably be moved out of this loop.